packages/concerto-vocabulary/lib/vocabularymanager.js

```diff
@@ -205,7 +205,18 @@ class VocabularyManager {
     resolveTerms(modelManager, namespace, locale, declarationName, propertyName) {
         const modelFile = modelManager.getModelFile(namespace);
         const classDecl = modelFile ? modelFile.getType(declarationName) : null;
-        const property = propertyName ? classDecl ? classDecl.getProperty(propertyName) : null : null;
+        let property;
+        if(classDecl && !classDecl.isScalarDeclaration()) {
+            if(classDecl.isMapDeclaration()) {
```

Comment:
Adding `isMapDeclaration` method to the `Declaration` class might help simplify this bit.

Reply:
Addressed and fixed.